DT-1240: add/remove custodian users from snapshot steward policy

pshapiro4broad opened the pull request:
__Jira ticket__: https://broadworkbench.atlassian.net/browse/DT-1240

## Addresses

When the `inherit steward` flag is set, we want to remove any custodian users that were present on a dataset snapshot. When the flag is unset, we want to re-add the custodian users.

## Summary of changes

- add a new step to support this operation
- add `email` property to the `SamPolicyModel` to avoid making two Sam calls to get the proxy email along with the policy members

## Testing Strategy

- unit tests

pshapiro4broad committed:
WIP: add/remove custodian users from snapshot steward policy

pshapiro4broad commented on src/main/resources/api/data-repository-openapi.yaml:
```diff
@@ -7170,6 +7170,8 @@ components:
       properties:
         name:
           type: string
+        email:
```
What'd really like to do here is remove the reliance on an API model type in an internal API response. But that would be a pretty big change on its own, as these APIs are used in many places (mostly tests).

pshapiro4broad committed:
spotless

pshapiro4broad committed:
Merge branch 'develop' into ps/dt-1240-inherit-steward-remove-users

pshapiro4broad committed:
unit tests

pshapiro4broad commented on src/main/java/bio/terra/service/dataset/flight/inheritsteward/AdjustStewardMembersStep.java:
```diff
@@ -0,0 +1,61 @@
+        Objects.requireNonNull(
+            inputParams.get(JobMapKeys.CUSTODIAN_USERS.getKeyName(), List.class));
+    AddRemoveApi api =
+        inheritSteward ? iamService::addPolicyMember : iamService::deletePolicyMember;
```
If we are at all concerned about performance, we may want to add special `IamService` APIs to handle this migration. Each call to these APIs makes two Sam calls; one to change the policy, and another to retrieve the current policy state which is unneeded here. Also note that each API call results in a new journal entry, although the journal is per-snapshot so that doesn't seem too noisy.

pshapiro4broad replied:
> Each call to these APIs makes two Sam calls; one to change the policy, and another to retrieve the current policy state which is unneeded here.

Looking at the code, it would be reasonable to make the callers who need the policy to explicitly call Sam separately from this API.

pshapiro4broad replied:
I have a PR onto this branch with this change. Or it could be merged in as a separate change after this PR is merged. https://github.com/DataBiosphere/jade-data-repo/pull/1939

fboulnois reviewed (approved):
overall looks reasonable 👍

pshapiro4broad committed:
change to internal Sam add/revoke policy APIs

Split out get operation from add/revoke policy operations. Also make policy typesafe in outer Sam APIs.

pshapiro4broad committed:
Merge branch 'develop' into ps/dt-1240-inherit-steward-remove-users
